packages/flutter/lib/src/rendering/sliver_list.dart

```diff
@@ -169,20 +169,22 @@ class RenderSliverList extends RenderSliverMultiBoxAdaptor {
     // If the scroll offset is at zero, we should make sure we are
     // actually at the beginning of the list.
     if (scrollOffset < precisionErrorTolerance) {
-      if (indexOf(firstChild) > 0) {
+      while (indexOf(firstChild) > 0) {
         final double earliestScrollOffset = childScrollOffset(firstChild);
         // We correct one child at a time. If there are more children before
         // the earliestUsefulChild, we will correct it once the scroll offset
-        // reach zero again.
+        // reaches zero again.
         earliestUsefulChild = insertAndLayoutLeadingChild(childConstraints, parentUsesSize: true);
         assert(earliestUsefulChild != null);
         final double firstChildScrollOffset = earliestScrollOffset - paintExtentOf(firstChild);
-        geometry = SliverGeometry(
-          scrollOffsetCorrection: -firstChildScrollOffset,
-        );
         final SliverMultiBoxAdaptorParentData childParentData = firstChild.parentData as SliverMultiBoxAdaptorParentData;
         childParentData.layoutOffset = 0.0;
-        return;
+        if (firstChildScrollOffset < -precisionErrorTolerance) {
```

Comment:
Can you add a comment to explain why we need to do this check and why we use the while loop? (something about the paint extent can be zero)

Reply:
Sure!